lib/Promise/functions.php

```diff
@@ -69,7 +71,9 @@ function ($reason) use ($fail) {
  * The returned promise will resolve or reject with the value or reason of
  * that first promise.
  *
- * @param Promise[] $promises
+ * @param Promise<int>[] $promises
+ *
+ * @return Promise<int>
```

Comment:
I guess this is a leftover?

```suggestion
 * @param Promise<mixed>[] $promises
 *
 * @return Promise<mixed>
```

same in other methods of this class?

Reply:
yes, leftover - fixing...